sylvia-derive/src/message.rs

```diff
@@ -118,26 +146,46 @@ impl<'a> StructMessage<'a> {
         let generics = emit_bracketed_generics(generics);
         let unused_generics = emit_bracketed_generics(unused_generics);
 
+        let base_name = match msg_ty {
+            MsgType::Instantiate => quote! { base_instantiate },
+            MsgType::Migrate => quote! { base_migrate },
+            _ => unimplemented!(),
+        };
+
         #[cfg(not(tarpaulin_include))]
         {
             quote! {
+                // This is the InstantiateMsg for the contract
                 #[allow(clippy::derive_partial_eq_without_eq)]
                 #[derive(#sylvia ::serde::Serialize, #sylvia ::serde::Deserialize, Clone, Debug, PartialEq, #sylvia ::schemars::JsonSchema)]
                 #[serde(rename_all="snake_case")]
                 pub struct #name #generics {
```

Comment:
Here, we create the overlying type, specific to the module type

Reply:
We could do better and use the blanket type that we have instead of creating a new struct. This could also be implemented in the #[sylvia::interface]  macro just like it's done for ContractEnumMessages.